lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -24,7 +24,7 @@ def self.check_visitor
   # Initializes the validator with a diagnostics producer. This object must respond to
   # `:will_accept?` and `:accept`.
   #
-  def initialize(diagnostics_producer)
+  def initialize(diagnostics_producer, label_provider)
```

Comment:
Nah, it is the diagnostics provider that has the label provider IIRC - should be no change to the checker's API. Look at where the diagnostics producer is coming from.
You can just use the standard label provider / diagnostics producer (I thing) since the new types are in the AST.